File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> seen;
        int left = 0;
        int longest = 0;

        for (int right = 0; right < s.length(); right++){
            while (seen.find(s[right]) != seen.end()){
                seen.erase(s[left]);
                left++;
            }
            seen.insert(s[right]);
            int window = right-left+1;
            longest = max(longest, window);
        }
        return longest;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        seen = set(nums)
        longest_seq = 0

        for num in nums:
            if num-1 not in seen:
                length = 1

                while num+length in seen:
                    length +=1
                longest_seq = max(length, longest_seq)
        
        return longest_seq
        